aten/src/ATen/TensorIterator.cpp

```diff
@@ -292,16 +292,30 @@ void TensorIteratorBase::reorder_dimensions() {
 // See the [Common Dtype Computation] note
 ScalarType TensorIteratorBase::compute_common_dtype() {
   at::native::ResultTypeState state = {};
+  std::ostringstream input_dtypes;
+
   for (const auto& op : operands_) {
     if (op.is_output) {
       continue;
     }
 
+    // Get input dtypes for error reporting.
+    if (!op.is_output) {
+      if (input_dtypes.tellp() == 0) {
+        input_dtypes << op.current_dtype;
+      } else {
+        input_dtypes << ", " << op.current_dtype;
+      }
+    }
+
     state = at::native::update_result_type_state(op.tensor(), state);
   }
 
   common_dtype_ = at::native::result_type(state);
-  TORCH_INTERNAL_ASSERT(common_dtype_ != ScalarType::Undefined);
+
+  TORCH_CHECK(
```

Comment:
there's no need to improve error message here, it was an internal assert and should remain an internal assert